### PR TITLE
Add JsonVersionConverter and tests

### PR DESCRIPTION
**Pull request by crobibero**
Backport not needed, as the removal was only targeting master.
Fixes https://github.com/jellyfin/jellyfin/issues/5379

### CODE REVIEW

File: MediaBrowser.Common/Json/Converters/JsonVersionConverter.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        /// <inheritdoc />
+        public override Version Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            => new Version(reader.GetString());
+
+        /// <inheritdoc />
+        public override void Write(Utf8JsonWriter writer, Version value, JsonSerializerOptions options)
+            => writer.WriteStringValue(value.ToString());
+    }
+}
```
Comment (dkanada) [inline MediaBrowser.Common/Json/Converters/JsonVersionConverter.cs:23]:
```suggestion
}

```

File: tests/Jellyfin.Common.Tests/Json/JsonVersionConverterTests.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        [Fact]
+        public void Serialize_Version_Success()
+        {
+            var input = new Version(1, 09, 59);
+            var output = "\"1.9.59\"";
+            var serializedInput = JsonSerializer.Serialize(input, _options);
+            Assert.Equal(output, serializedInput);
+        }
+    }
+}
```
Comment (dkanada) [inline tests/Jellyfin.Common.Tests/Json/JsonVersionConverterTests.cs:36]:
```suggestion
}

```

**Comment by Bond-009** [comment]
What does it currently output?

**Comment by crobibero** [comment]
> What does it currently output?

Something like 
`{“Major”: 1, “Minor”: 9, “Build”: 59, [...]}`

**Review by BaronGreenback — APPROVED** [review]

File: MediaBrowser.Common/Json/Converters/JsonVersionConverter.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    /// </remarks>
+    public class JsonVersionConverter : JsonConverter<Version>
+    {
+        /// <inheritdoc />
+        public override Version Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            => new Version(reader.GetString());
+
+        /// <inheritdoc />
+        public override void Write(Utf8JsonWriter writer, Version value, JsonSerializerOptions options)
+            => writer.WriteStringValue(value.ToString());
```
Comment (Bond-009) [inline MediaBrowser.Common/Json/Converters/JsonVersionConverter.cs:21]: Could possibly prevent a string alloc here by doing a stack alloc and using `Version.TryFormat`

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**